nsqd/diskqueue.go

```diff
@@ -261,6 +261,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
 		return nil, err
 	}
 
+	if msgSize < minMsgSize {
```

Comment:
If the receiver reads an invalid message, there is no way for the receiver
to invalidate the current backing file. This also means that any new read
from the current file cannot be trusted, as if the reader trusts the file
to only contain messages, and it doesn't, how can the reader know any more
of the current file is valid?

On Saturday, July 11, 2015, Matt Reiferson <notifications@github.com> wrote:

> In nsqd/diskqueue.go
> <https://github.com/bitly/nsq/pull/600#discussion_r34415836>:
>
> > @@ -261,6 +261,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
> >  		return nil, err
> >  	}
> >
> > +	if msgSize < minMsgSize {
>
> DiskQueue doesn't (and arguably shouldn't) know about "messages". I think
> that DiskQueue should test for msgSize > 0, not minMsgSize. The receiver
> of the read channel should enforce that the size is appropriate for a
> message, i.e. at least minMsgSize, so we can create an additional check
> there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitly/nsq/pull/600/files#r34415836>.
>